src/quest/historian.py

```diff
@@ -952,71 +951,29 @@ async def get_resources(self, identity):
         if self._fatal_exception.done():
             await self._fatal_exception
 
-        # Get public resources first
-        resources = {
-            entry['name']: entry['resource']
-            for entry in self._resources.values()
-            if entry['identity'] is None
-        }
-
-        # Now add in (and override) identity-specific resources
+        # Return a dictionary of resources wrapped in register_external_event wrappers
+        resources = {}
         for entry in self._resources.values():
-            if entry['identity'] == identity:
-                resources[entry['name']] = entry['resource']
-
-        # Now wrap the resources in register_external_event wrappers
-        return {k: wrap_methods_as_historian_events(res, k, identity, self, internal=False) for k, res in
-                resources.items()}
-
-    async def stream_resources(self, identity):
-        """
-        Provide a stream of resource snapshots for this workflow.
-        Everytime the workflow resource state changes, an update will be published.
-
-        NOTE: Once you start the resource stream,
-        the workflow will not progress unless you iterate this stream.
-        Use wait_for_completion() to automatically iterate (and ignore) the remaining updates.
-        """
-
-        # The caller of this function lives outside the step management of the historian
-        #         -> don't replay, just yield event changes in realtime
-
-        logging.debug(f'Resource stream created for {identity}')
-
-        # _handle_resource_updates puts on this queue when the resources have changed
-        self._resource_queues.setdefault(identity, asyncio.Queue())
-
-        # this event ensures that the historian yields to whoever is using the
-        # resources before continuing on
-        # (otherwise the historian would move until blocked,
-        # and the observer wouldn't get a chance to respond to resource changes)
-        self._resource_events.setdefault(identity, asyncio.Event())
-
-        # Yield the current resources immediately
-        yield await self.get_resources(identity)
-
-        while not self._workflow_completed:
-            # Yield new resources updates as they become available
-            await self._resource_queues[identity].get()
-            yield await self.get_resources(identity)
-            self._resource_events[identity].set()
+            # Always return public resources and any private resources that match the identity
+            if entry['identity'] is None or entry['identity'] == identity:
+                resources[entry['name']] = wrap_methods_as_historian_events(
+                    entry['resource'],
+                    entry['name'],
+                    entry['identity'],
+                    self,
+                    internal=False
+                )
```

Comment:
We need a test: if the private identity creates a resource, then the public identity creates a resource by the same name, we should still get the private version.

I'm guessing this code fails that requirement.

We probably need two loops - first the public, then the private loop (so the private resources override the public ones).

Reply:
I confirmed that this is in fact the case. When there is a public resource and private resource of the same name, the public version is returned to a private caller. I will refactor the code here

Reply:
Okay there is some weirdness here that we need to discuss. When changing this so that the private listener only receives the private version of the resource, the problem is that anytime an update occurs to either of the resources with the same name, the private listener will receive an "updated" view of resources containing the private resource. This update happens and returns the private resource, even if no update to the private resource occurred. So maybe we override only if the update itself is tied to the private identity? Either way there is still weirdness because ultimately only one resource is returned in the resource view to the listener even though there is two...

Reply:
Good catch.

I think we need to return resources with associated identities and leave it to the caller to decide which resource it wants to use (or how it wants to use them). 

I have in my mind the possibility of adding identity groups to quest - like roles in Discord - where the client can subscribe to all resources owned by the group (public being the ultimate group). In this case it also makes sense that the client would know the specific identities to which a resource belongs. 

So, instead of returning a dictionary of `name: resource` pairs, we may need `(name, identity): resource` pairs.